src/preload/overrides.c

```diff
@@ -179,44 +179,55 @@ void spurious_desched_syscall(struct syscall_info* info) {
   impose_spurious_desched = 0;
 }
 
-#ifndef __aarch64__
-
-/**
- * glibc geteuid() can be compiled to instructions ending in "syscall; ret"
- * which sometimes can't be hooked. So override it here with something that
- * can be hooked.
- * This is not an issue on aarch64 since we only need to patch a single instruction.
- */
-uid_t geteuid(void) {
-#ifdef __i386__
-  return syscall(SYS_geteuid32);
-#else
-  return syscall(SYS_geteuid);
-#endif
-}
-
 /**
  * clang's LeakSanitizer has regular threads call sched_yield() in a loop while
  * a helper thread ptrace-attaches to them. If we let sched_yield() enter the
  * syscallbuf, the helper thread sees that the regular thread SP register
  * is pointing to the syscallbuf alt-stack, outside the stack region it
  * expects, which causes it to freak out.
  * So, override sched_yield() to perform the syscall in a way that can't
- * be syscall-buffered. (We have no syscall hook for `syscall` followed by
- * `inc %ecx`).
+ * be syscall-buffered.
  */
 int sched_yield(void) {
-  int trash;
 #ifdef __i386__
+  // We have no syscall hook for `syscall` followed by `inc %ecx`
+  int trash;
   asm volatile ("int $0x80; inc %0" : "=c"(trash) : "a"(SYS_sched_yield));
 #elif defined(__x86_64__)
+  // We have no syscall hook for `syscall` followed by `inc %ecx`
+  int trash;
   asm volatile ("syscall; inc %0" : "=c"(trash) : "a"(SYS_sched_yield));
+#elif defined(__aarch64__)
+  register long x8 __asm__("x8") = SYS_sched_yield;
+  // We explicitly blacklisted syscall that follows `mov x8, 0xdc`
+  // to avoid patching clone. Abuse that to prevent this from being patched.
```

Comment:
I see that it's useful to have an instruction sequence that doesn't get patched, but if we only need that then it would be cleaner to check for a special instruction following `svc`.

Reply:
This actually comes **after** the pattern matching. I was going to add a check for a strange instruction following svc but realize that I could abuse the currently one so I didn't bother adding anything......